traffic_ops/traffic_ops_golang/routing/routes.go

```diff
@@ -104,6 +105,9 @@ func Routes(d ServerData) ([]Route, []RawRoute, http.Handler, error) {
 		{1.1, http.MethodPost, `asns/?$`, api.CreateHandler(&asn.TOASNV11{}), auth.PrivLevelOperations, Authenticated, nil},
 		{1.1, http.MethodDelete, `asns/{id}$`, api.DeleteHandler(&asn.TOASNV11{}), auth.PrivLevelOperations, Authenticated, nil},
 
+		// Traffic Stats access
+		{1.2, http.MethodGet, `deliveryservice_stats`, trafficstats.GetDSStats, auth.PrivLevelOperations, Authenticated, nil},
```

Comment:
this should be 1.1 since you are overwriting the perl entirely. otherwise, 1.2 would serve Go and 1.1 would continue to serve Perl.

Reply:
is operations really the role enforced in Perl? seems high.

Reply:
also, at one time this routes.go file was alphabetically organized. not sure what happened to that.

Reply:
Alphabetical organization might be dangerous since the order matters w.r.t. the API version and regex matches. I.e. the same route for different minor versions needs to have the minor versions in descending order (since a 1.1 request will not match a 1.3 or 1.2 route, but a 1.3 request will match a 1.2 or 1.1 route). There may be some regexes that are overlapping as well, so the _longer_ regexes should come before the shorter regexes so that the best match is taken.

Reply:
The route doesn't exist prior to 1.2 - if you request it today at 1.1 you'll get a 404.

I'm fairly certain the Perl used Ops, but I agree that that's too high. However, I think I'm changing a lot as it is.

Reply:
fairly certain? better check. :)

Reply:
> Alphabetical organization might be dangerous 

oh yeah, i forgot. there are some in there that are very order dependent. but if adding a `d` one, might as well put it  next to the `d`s imo. kinda sucks that they can't be alphabetized somehow, gives me a headache searching thru for stuff that i forgot the name to.

Reply:
@ocket8888 - this is the current check i guess from DeliveryServiceStats.pm

```
			if ( $self->is_delivery_service_name_assigned($ds_name) || &is_admin($self) || &is_oper($self) ) {
```
if it's assigned to your or you're admin or ops. (not sure why it doesn't just check ops...but anyhow)

so this is broken and i think you should fix it. the real check should be:
1. do you pass the tenancy check?
2. are you readonly or above?

Reply:
so basically the way it is implemented now it is asking "is this ds assigned to you" and if so, you are good to go. but that behavior should have been changed a long time ago to use tenancy instead.

in which case NO role was checked but i think R/O makes perfect sense here.

Reply:
I agree, but the goal of a rewrite is usually to change as little as possible, so if anything I almost feel like I should check for a user-ds assignment instead.

Reply:
I smell a tech debt issue in your future :)